pennylane/qchem/openfermion_obs.py

```diff
@@ -1123,18 +1122,36 @@ def _molecular_hamiltonian(
             h = qml.map_wires(h, wires_map)
         return h, 2 * len(active)
 
-    if method == "pyscf" and mapping.strip().lower() == "jordan_wigner":
+    if method == "pyscf":
         core_constant, one_mo, two_mo = _pyscf_integrals(
             symbols, geometry_hf, charge, mult, basis, active_electrons, active_orbitals
         )
 
         hf = qml.qchem.fermionic_observable(core_constant, one_mo, two_mo)
 
         if active_new_opmath():
-            h_pl = qml.jordan_wigner(hf, wire_map=wires_map, tol=1.0e-10).simplify()
-
+            if mapping.strip().lower() == "jordan_wigner":
+                h_pl = qml.jordan_wigner(hf, wire_map=wires_map, tol=1.0e-10).simplify()
+            elif mapping.strip().lower() == "parity":
+                qubits = len(hf.wires)
+                h_pl = qml.parity_transform(hf, qubits, wire_map=wires_map, tol=1.0e-10).simplify()
+            elif mapping.strip().lower() == "bravyi_kitaev":
+                qubits = len(hf.wires)
+                h_pl = qml.bravyi_kitaev(hf, qubits, wire_map=wires_map, tol=1.0e-10).simplify()
         else:
-            h_pl = qml.jordan_wigner(hf, ps=True, wire_map=wires_map, tol=1.0e-10).hamiltonian()
+            if mapping.strip().lower() == "jordan_wigner":
+                h_pl = qml.jordan_wigner(hf, ps=True, wire_map=wires_map, tol=1.0e-10).hamiltonian()
+            elif mapping.strip().lower() == "parity":
+                qubits = len(hf.wires)
+                h_pl = qml.parity_transform(
+                    hf, qubits, ps=True, wire_map=wires_map, tol=1.0e-10
+                ).hamiltonian()
+            elif mapping.strip().lower() == "bravyi_kitaev":
+                qubits = len(hf.wires)
+                h_pl = qml.bravyi_kitaev(
+                    hf, qubits, ps=True, wire_map=wires_map, tol=1.0e-10
+                ).hamiltonian()
```

Comment:
This long if else block contains a lot of repeated work and could be simplified to be made more readable. I think it's best to define things like mapping.strip.lower() or the # of qubits once and use it throughout the if-else block.

Reply:
I have cleaned it up some, but I don't want to use ternary operators as that can make code hard to read with nested if statements.